addons/website/static/src/snippets/s_map/options.js

```diff
@@ -1,52 +1,31 @@
-odoo.define('options.s_map_options', function (require) {
-'use strict';
+/** @odoo-module **/
 
 const {_t} = require('web.core');
 const options = require('web_editor.snippets.options');
 
 options.registry.Map = options.Class.extend({
     /**
-     *
      * @override
      */
     onBuilt: function () {
-        this.updateUI();
+        this._updateSource();
     },
 
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
     /**
-     * @override
+     *
+     * @see this.selectClass for parameters
      */
-    updateUI() {
-        const dataset = this.$target[0].dataset;
-        const embedded = this.$target.find('.o_embedded');
-        const info = this.$target.find('.missing_option_warning');
-        if (dataset.mapAddress) {
-            const url = 'https://maps.google.com/maps?q=' + encodeURIComponent(dataset.mapAddress)
-                + '&t=' + encodeURIComponent(dataset.mapType)
-                + '&z=' + encodeURIComponent(dataset.mapZoom)
-                + '&ie=UTF8&iwloc=&output=embed';
-            if (url !== embedded.attr('src')) {
-                embedded.attr('src', url);
-            }
-            embedded.removeClass('d-none');
-            info.addClass('d-none');
-        } else {
-            embedded.attr('src', 'about:blank');
-            embedded.addClass('d-none');
-            info.removeClass('d-none');
-        }
-        if (! (this.$target.hasClass('o_half_screen_height') || this.$target.hasClass('o_full_screen_height'))) {
-            this.$target.css('min-height', this.$target[0].dataset.mapMinHeight);
+    selectDataAttribute: function (previewMode, widgetValue, params) {
+        this._super.apply(this, arguments);
```

Comment:
As all option methods are async, you should await the `_super` call. Also, use ES6 form when you can:

```javascript
await this._super(...arguments);
```